plugins/pulumi/pulumi_access_token.go

```diff
@@ -109,3 +111,31 @@ func TryPulumiConfigFile() sdk.Importer {
 		}
 	})
 }
+
+// DeleteSecretsFileProvisioner removes the Pulumi credentials file before letting the real provisioner do its magic.
+type DeleteSecretsFileProvisioner struct {
+	sdk.Provisioner
+
+	RealProvisioner sdk.Provisioner
+}
+
+func DeleteSecretsFile(provisioner sdk.Provisioner) sdk.Provisioner {
+	return DeleteSecretsFileProvisioner{
+		RealProvisioner: provisioner,
+	}
+}
+
+func (p DeleteSecretsFileProvisioner) Provision(ctx context.Context, in sdk.ProvisionInput, out *sdk.ProvisionOutput) {
+	// delete the current credentials file.
+	os.Remove(fmt.Sprintf("%s/%s", in.HomeDir, ".pulumi/credentials.json"))
```

Comment:
As discussed over our call, this won't work nicely with the security guarantees 1Password is trying to provide through the shell plugins. We are aiming to not allow any secrets on disk, and with the current approach secrets will remain on disk for a long time. So the plugin doesn't bring any security value. 

A better approach is to use the `Deprovision` function to remove the file, but we'll first need to make the deprovisioning logic available in the CLI. This still wouldn't be acceptable the the plaintext secrets will still touch the disk.

We are researching ways of better supporting temp file provisioning.